lib/statsd/instrument.rb

```diff
@@ -46,6 +46,21 @@ def self.generate_metric_name(metric_name, callee, *args)
       metric_name.respond_to?(:call) ? metric_name.call(callee, args).gsub('::', '.') : metric_name.gsub('::', '.')
     end
 
+    # @private
+    def self.process_metric_options(metric_options, calle, *method_args)
+      metric_options.map do |m_value|
+        next m_value unless m_value.is_a?(Hash)
+
+        m_value.each_with_object({}) do |(name, value), obj|
+          if value.respond_to?(:call)
+            obj[name] = value.call(calle, method_args)
```

Comment:
This library gives a strong guarantee that it never raises exceptions. We should really be catching exception here.